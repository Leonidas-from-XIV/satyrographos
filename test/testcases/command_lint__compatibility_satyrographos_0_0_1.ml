open Core
open Satyrographos_testlib

let satysfi_package_opam =
  "satysfi-package.opam", TestLib.opam_file_for_test
    ~name:"satysfi-package"
    ~version:"0.1"
    ()

let satyristes =
  "Satyristes", sprintf
    {|(version "0.0.2")

(library
  (name "package")
  (version "0.1")
  (sources ((package "test.satyh" "test.satyh")))
  (opam "satysfi-package.opam")
  (dependencies ())
  (compatibility ((satyrographos 0.0.1))))
|}

let test_satyh =
  "test.satyh", {||}

let opam_libs = Satyrographos.Library.[
    {empty with
     name = Some "package";
     files = LibraryFiles.of_alist_exn [
         "packages/package/test.satyh", `Content ""
       ]
    };
]

let files =
  [ satysfi_package_opam;
    satyristes;
    test_satyh;
  ]

let () =
  TestCommand.test_lint_command ~opam_libs files
