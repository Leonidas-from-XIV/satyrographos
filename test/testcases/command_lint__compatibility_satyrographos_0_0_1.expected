@@test_library@@/Satyristes:3:1: (module package):
Warning: lib/compat/0.0.1
  Compatibility warnings for Satyrographos 0.0.1 libraries are no longer effective.

1 problem(s) found.
